vllm/compilation/activation_quant_fusion.py

```diff
@@ -80,42 +82,39 @@ class SiluMulFp8StaticQuantPattern(ActivationQuantPattern):
     Fusion for SiluMul+Fp8StaticQuant Pattern
     """
 
-    def __init__(self, symmetric: bool = True):
-        quant_key = QuantKey(
-            dtype=FP8_DTYPE, scale=kStaticTensorScale, symmetric=symmetric
-        )
-        super().__init__(quant_key)
+    def __init__(self):
+        super().__init__(kFp8StaticTensorSym)
+        self.quant_matcher = MatcherQuantFP8(kFp8StaticTensorSym)
 
     def register(self, pm_pass: PatternMatcherPass):
         def pattern(
-            result: torch.Tensor,
-            result_silu_mul: torch.Tensor,
             input: torch.Tensor,
             scale: torch.Tensor,
         ):
-            at1 = auto_functionalized(SILU_MUL_OP, result=result_silu_mul, input=input)
-            at2 = auto_functionalized(
-                self.QUANT_OP, result=result, input=at1[1], scale=scale
-            )
-            return at2[1]
+            result_silu_mul = self.silu_and_mul_matcher(input)
+            result_quant = self.quant_matcher(result_silu_mul, scale)
+            return result_quant[0]
 
         def replacement(
-            result: torch.Tensor,
-            result_silu_mul: torch.Tensor,
             input: torch.Tensor,
             scale: torch.Tensor,
         ):
+            d = input.shape[-1] // 2
+            output_shape = input.shape[:-1] + (d,)
+            result = torch.empty(
+                output_shape, device=input.device, dtype=self.quant_dtype
+            )
             at = auto_functionalized(
                 self.FUSED_OP, result=result, input=input, scale=scale
             )
             return at[1]
 
         inputs = [
-            self.empty_quant(5, 4),  # result
-            empty_bf16(5, 4),  # result_silu_mul
-            empty_bf16(5, 4),  # input
-            empty_fp32(1, 1),  # scale
+            # input, weight
```

Comment:
This should be just input, no?

```suggestion
            # input
```